rest_framework-stubs/routers.pyi

```diff
@@ -12,56 +13,56 @@ from rest_framework.viewsets import ViewSetMixin
 
 class Route(NamedTuple):
     url: str
-    mapping: Dict[str, str]
+    mapping: dict[str, str]
     name: str
     detail: bool
-    initkwargs: Dict[str, Any]
+    initkwargs: dict[str, Any]
 
 class DynamicRoute(NamedTuple):
     url: str
     name: str
     detail: bool
-    initkwargs: Dict[str, Any]
+    initkwargs: dict[str, Any]
 
 def escape_curly_brackets(url_path: str) -> str: ...
 def flatten(list_of_lists: Iterable[Iterable[Any]]) -> Iterable[Any]: ...
 
 class RenameRouterMethods(RenameMethodsBase):
-    renamed_methods: Iterable[Union[str, Callable]] = ...
+    renamed_methods: Iterable[str | Callable]
 
 class BaseRouter(metaclass=RenameRouterMethods):
-    registry: List[Tuple[str, Type[ViewSetMixin], str]]
+    registry: list[tuple[str, type[ViewSetMixin], str]]
     def register(
-        self, prefix: str, viewset: Type[ViewSetMixin], basename: Optional[str] = ..., base_name: Optional[str] = ...
+        self, prefix: str, viewset: type[ViewSetMixin], basename: str | None = ..., base_name: str | None = ...
     ) -> None: ...
-    def get_default_basename(self, viewset: Type[ViewSetMixin]) -> str: ...
-    def get_urls(self) -> List[_AnyURL]: ...
+    def get_default_basename(self, viewset: type[ViewSetMixin]) -> str: ...
+    def get_urls(self) -> list[_AnyURL]: ...
     @property
-    def urls(self) -> List[_AnyURL]: ...
+    def urls(self) -> list[_AnyURL]: ...
 
 class SimpleRouter(BaseRouter):
-    routes: List[Union[Route, DynamicRoute]] = ...
+    routes: list[Route | DynamicRoute]
     trailing_slash: str
     def __init__(self, trailing_slash: bool = ...) -> None: ...
-    def get_routes(self, viewset: Type[ViewSetMixin]) -> List[Route]: ...
+    def get_routes(self, viewset: type[ViewSetMixin]) -> list[Route]: ...
     def _get_dynamic_route(self, route: DynamicRoute, action: Any) -> Route: ...
-    def get_method_map(self, viewset: Type[ViewSetMixin], method_map: Mapping[str, str]) -> Dict[str, str]: ...
-    def get_lookup_regex(self, viewset: Type[ViewSetMixin], lookup_prefix: str = ...) -> str: ...
+    def get_method_map(self, viewset: type[ViewSetMixin], method_map: Mapping[str, str]) -> dict[str, str]: ...
+    def get_lookup_regex(self, viewset: type[ViewSetMixin], lookup_prefix: str = ...) -> str: ...
 
 class APIRootView(views.APIView):
-    _ignore_model_permissions: bool = ...
-    api_root_dict: Optional[Dict[str, str]] = ...
+    _ignore_model_permissions: bool
+    api_root_dict: dict[str, str] | None
     def get(self, request: Request, *args: Any, **kwargs: Any) -> Response: ...
 
 class DefaultRouter(SimpleRouter):
-    include_root_view: bool = ...
-    include_format_suffixes: bool = ...
-    root_view_name: str = ...
-    default_schema_renderers = None
+    include_root_view: bool
+    include_format_suffixes: bool
+    root_view_name: str
+    default_schema_renderers = None  # noqa: Y026
```

Comment:
I think `None` is the default value and should be removed.

You can type this as `: Any` for now.

Reply:
But we can fix these in a follow-up just as well. Up to you.